src/lit-html.ts

```diff
@@ -12,21 +12,39 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
+/**
+ * TypeScript has a problem with precompiling templates literals
+ * https://github.com/Microsoft/TypeScript/issues/17956
+ */
+const shouldCacheTemplates = ((_t: any) => _t() === _t())(() => ((s: any) => s)``);
```

Comment:
I would name this `cachesTemplates`, as in "the browser caches template literals properly"